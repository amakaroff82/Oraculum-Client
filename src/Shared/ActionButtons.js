import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import Button from 'material-ui/Button';

const styles = theme => ({
  button: {
    margin: theme.spacing.unit,
  },
});

export const ConditionalButton = ({
  buttonProps: { textId, textValues, ...buttonProps },
  condition,
  ...rest
}) =>
  (condition === undefined || condition) && buttonProps ? (
    <Button {...buttonProps} {...rest}>
      "text val"
    </Button>
  ) : null;

ConditionalButton.propTypes = {
  buttonProps: PropTypes.shape({
    textId: PropTypes.string,
  }),
  condition: PropTypes.bool,
};

ConditionalButton.displayName = 'ConditionalButton';

export const ActionButtons = ({
  classes,
  primaryButtonProps: {
    condition: primaryButtonCondition,
    ...primaryButtonProps
  } = { condition: false },
  cancelButtonProps: {
    condition: cancelButtonCondition,
    ...cancelButtonProps
  } = { condition: false },
  extraButtons,
  children,
}) => (
  <div>
    <ConditionalButton
      raised
      className={classes.button}
      buttonProps={{ textId: 'cancel', ...cancelButtonProps }}
      condition={cancelButtonCondition}
    />
    {extraButtons &&
      extraButtons.map(({ condition, ...buttonProps }, index) => (
        <ConditionalButton
          key={index}
          raised
          className={classes.button}
          buttonProps={buttonProps}
          condition={condition}
        />
      ))}
    <ConditionalButton
      raised
      className={classes.button}
      buttonProps={primaryButtonProps}
      condition={primaryButtonCondition}
      color="primary"
    />
    {children}
  </div>
);

const ButtonPropsPropType = PropTypes.shape({
  condition: PropTypes.bool,
});

ActionButtons.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string),
  primaryButtonProps: ButtonPropsPropType,
  cancelButtonProps: ButtonPropsPropType,
  extraButtons: PropTypes.arrayOf(ButtonPropsPropType),
  children: PropTypes.node,
};

ActionButtons.displayName = 'ActionButtons';

export default withStyles(styles, { name: 'ActionButtons' })(ActionButtons);
