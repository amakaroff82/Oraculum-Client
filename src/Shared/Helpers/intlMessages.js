export const intlMessages = {
  orchestration: { id: 'Orchestration', defaultMessage: 'Orchestration' },
  products: { id: 'Products', defaultMessage: 'Products' },
  status: { id: 'Status', defaultMessage: 'Status' },
  dateLastModified: {
    id: 'DateLastModified',
    defaultMessage: 'Date last modified',
  },
  title: { id: 'Title', defaultMessage: 'Titre' },
  topic: { id: 'Topic', defaultMessage: 'Topic' },
  createdBy: { id: 'CreatedBy', defaultMessage: 'Created by' },
  version: { id: 'Version', defaultMessage: 'Version' },
};

export default intlMessages;
