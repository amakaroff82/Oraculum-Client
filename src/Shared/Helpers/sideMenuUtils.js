import React from 'react';
import { ListItem, ListItemText } from 'material-ui/List';

export const renderListItems = (listItemData, listAction, identifierKey) => {
  if (listItemData && listAction && identifierKey) {
    return listItemData.map((element, index) => {
      return (
        <ListItem
          key={element[identifierKey]}
          button
          onClick={() => listAction(element)}
        >
          <ListItemText color="primary" primary={element[identifierKey]} />
        </ListItem>
      );
    });
  }
};
