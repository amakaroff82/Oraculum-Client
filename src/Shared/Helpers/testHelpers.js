// Create mock response for testing
export const makeMockResponse = (
  response,
  status = 200,
  statusText = 'OK',
  headers = { 'Content-Type': 'application/json' }
) => {
  const init = {
    status,
    statusText,
    headers,
  };
  return new window.Response(JSON.stringify(response), init);
};

export const jsonParseError = () => {
  try {
    JSON.parse('+data+');
  } catch (err) {
    return err;
  }
};

export const makeFakeError = (
  response = null,
  status = 400,
  statusText = 'Bad Request'
) => {
  const error = new Error('Error');
  error.response = makeMockResponse(response, status, statusText);
  return error;
};
