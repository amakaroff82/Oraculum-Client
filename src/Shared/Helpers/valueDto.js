/// Create items for binding Std ValueDTOs from the web service
export const getDropDownValues = dtoArray => {
  if (!dtoArray) {
    return [];
  }
  return dtoArray.map(element => ({
    value: element.code,
    label: element.description || element.code,
  }));
};

export const getAutocompleteValues = dtoArray => {
  if (!dtoArray) {
    return [];
  }
  var result = {};
  for (var value of dtoArray) {
    result[value['code']] = value['description'] || value['code'];
  }
  return result;
};
