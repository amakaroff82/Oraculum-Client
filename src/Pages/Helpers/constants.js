import { MC_BASE_URL } from '../../Shared/constants';

