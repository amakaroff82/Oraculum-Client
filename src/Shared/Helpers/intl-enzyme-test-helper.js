/**
 * Components using the react-intl module require access to the intl context.
 * This is not available when mounting single components in Enzyme.
 * These helper functions aim to address that and wrap a valid,
 * English-locale intl context around them.
 */

import React from 'react';
import { mount, render, shallow } from 'enzyme';
import { intlMessages } from './intlMessages';

