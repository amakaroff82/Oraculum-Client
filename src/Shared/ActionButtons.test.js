import React from 'react';
import { shallow } from 'enzyme';
import { mountWithIntl } from '../Shared/Helpers/intl-enzyme-test-helper';
import toJson from 'enzyme-to-json';
import { ActionButtons, ConditionalButton } from './ActionButtons';
import Button from 'material-ui/Button';

const fakePrimaryButtonProps = { textId: 'save' };
const fakeCancelButtonProps = { onClick: jest.fn() };

const defaultProps = {
  classes: {},
};

const setup = setupProps => {
  const allProps = { ...defaultProps, ...setupProps };
  return shallow(<ActionButtons {...allProps} />);
};

describe('<ActionButtons />', () => {
  it('renders primary button without cancel button', () => {
    const buttons = setup({ primaryButtonProps: fakePrimaryButtonProps });
    expect(toJson(buttons)).toMatchSnapshot();
  });

  it('renders cancel button without primary button', () => {
    const buttons = setup({ cancelButtonProps: fakeCancelButtonProps });
    expect(toJson(buttons)).toMatchSnapshot();
  });

  it('renders cancel button before primary button', () => {
    const buttons = setup({
      primaryButtonProps: fakePrimaryButtonProps,
      cancelButtonProps: fakeCancelButtonProps,
    });

    expect(
      buttons
        .find(ConditionalButton)
        .at(0)
        .prop('buttonProps').onClick
    ).toEqual(fakeCancelButtonProps.onClick);
    expect(
      buttons
        .find(ConditionalButton)
        .at(1)
        .prop('buttonProps').textId
    ).toEqual(fakePrimaryButtonProps.textId);
    expect(toJson(buttons)).toMatchSnapshot();
  });

  it('renders with extraButtons', () => {
    const buttons = setup({
      primaryButtonProps: fakePrimaryButtonProps,
      cancelButtonProps: fakeCancelButtonProps,
      extraButtons: [{ textId: 'extra1' }, { textId: 'extra2' }],
    });

    expect(toJson(buttons)).toMatchSnapshot();
  });
});

describe('<ConditionalButton />', () => {
  it('does not render when condition is falsy', () => {
    expect(
      shallow(
        <ConditionalButton
          condition={false}
          buttonProps={fakePrimaryButtonProps}
        />
      ).getNode()
    ).toBeNull();
  });

  it('renders when condition is undefined', () => {
    expect(
      shallow(<ConditionalButton buttonProps={fakePrimaryButtonProps} />).find(
        Button
      )
    ).toHaveLength(1);
  });

  it('renders expected button', () => {
    expect(
      toJson(
        mountWithIntl(
          <ConditionalButton buttonProps={fakePrimaryButtonProps} />
        )
      )
    ).toMatchSnapshot();
  });
});
