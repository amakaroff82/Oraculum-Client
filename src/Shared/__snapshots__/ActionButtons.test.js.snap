// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionButtons /> renders cancel button before primary button 1`] = `
<div>
  <ConditionalButton
    buttonProps={
      Object {
        "onClick": [Function],
        "textId": "cancel",
      }
    }
    raised={true}
  />
  <ConditionalButton
    buttonProps={
      Object {
        "textId": "save",
      }
    }
    color="primary"
    raised={true}
  />
</div>
`;

exports[`<ActionButtons /> renders cancel button without primary button 1`] = `
<div>
  <ConditionalButton
    buttonProps={
      Object {
        "onClick": [Function],
        "textId": "cancel",
      }
    }
    raised={true}
  />
  <ConditionalButton
    buttonProps={Object {}}
    color="primary"
    condition={false}
    raised={true}
  />
</div>
`;

exports[`<ActionButtons /> renders primary button without cancel button 1`] = `
<div>
  <ConditionalButton
    buttonProps={
      Object {
        "textId": "cancel",
      }
    }
    condition={false}
    raised={true}
  />
  <ConditionalButton
    buttonProps={
      Object {
        "textId": "save",
      }
    }
    color="primary"
    raised={true}
  />
</div>
`;

exports[`<ActionButtons /> renders with extraButtons 1`] = `
<div>
  <ConditionalButton
    buttonProps={
      Object {
        "onClick": [Function],
        "textId": "cancel",
      }
    }
    raised={true}
  />
  <ConditionalButton
    buttonProps={
      Object {
        "textId": "extra1",
      }
    }
    raised={true}
  />
  <ConditionalButton
    buttonProps={
      Object {
        "textId": "extra2",
      }
    }
    raised={true}
  />
  <ConditionalButton
    buttonProps={
      Object {
        "textId": "save",
      }
    }
    color="primary"
    raised={true}
  />
</div>
`;

exports[`<ConditionalButton /> renders expected button 1`] = `
<ConditionalButton
  buttonProps={
    Object {
      "textId": "save",
    }
  }
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
    }
  }
>
  <withStyles(Button)
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {},
        "now": [Function],
        "textComponent": "span",
      }
    }
  >
    <Button
      classes={
        Object {
          ".MuiButton-fab-14:active": "MuiButton-fab-14:active",
          ".MuiButton-flatAccent-5:hover": "MuiButton-flatAccent-5:hover",
          ".MuiButton-flatContrast-6:hover": "MuiButton-flatContrast-6:hover",
          ".MuiButton-flatPrimary-4:hover": "MuiButton-flatPrimary-4:hover",
          ".MuiButton-raised-8.MuiButton-disabled-13": "MuiButton-raised-8.MuiButton-disabled-13",
          ".MuiButton-raised-8.MuiButton-keyboardFocused-9": "MuiButton-raised-8.MuiButton-keyboardFocused-9",
          ".MuiButton-raised-8:active": "MuiButton-raised-8:active",
          ".MuiButton-raised-8:hover": "MuiButton-raised-8:hover",
          ".MuiButton-raised-8:hover.MuiButton-disabled-13": "MuiButton-raised-8:hover.MuiButton-disabled-13",
          ".MuiButton-raisedAccent-11:hover": "MuiButton-raisedAccent-11:hover",
          ".MuiButton-raisedPrimary-10:hover": "MuiButton-raisedPrimary-10:hover",
          ".MuiButton-root-1:hover": "MuiButton-root-1:hover",
          ".MuiButton-root-1:hover.MuiButton-disabled-13": "MuiButton-root-1:hover.MuiButton-disabled-13",
          "colorInherit": "MuiButton-colorInherit-7",
          "dense": "MuiButton-dense-2",
          "disabled": "MuiButton-disabled-13",
          "fab": "MuiButton-fab-14",
          "flatAccent": "MuiButton-flatAccent-5",
          "flatContrast": "MuiButton-flatContrast-6",
          "flatPrimary": "MuiButton-flatPrimary-4",
          "keyboardFocused": "MuiButton-keyboardFocused-9",
          "label": "MuiButton-label-3",
          "raised": "MuiButton-raised-8",
          "raisedAccent": "MuiButton-raisedAccent-11",
          "raisedContrast": "MuiButton-raisedContrast-12",
          "raisedPrimary": "MuiButton-raisedPrimary-10",
          "root": "MuiButton-root-1",
        }
      }
      color="default"
      dense={false}
      disableFocusRipple={false}
      disableRipple={false}
      disabled={false}
      fab={false}
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": Object {},
          "now": [Function],
          "textComponent": "span",
        }
      }
      raised={false}
      type="button"
    >
      <withStyles(ButtonBase)
        className="MuiButton-root-1"
        disableRipple={false}
        disabled={false}
        focusRipple={true}
        intl={
          Object {
            "defaultFormats": Object {},
            "defaultLocale": "en",
            "formatDate": [Function],
            "formatHTMLMessage": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatPlural": [Function],
            "formatRelative": [Function],
            "formatTime": [Function],
            "formats": Object {},
            "formatters": Object {
              "getDateTimeFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralFormat": [Function],
              "getRelativeFormat": [Function],
            },
            "locale": "en",
            "messages": Object {},
            "now": [Function],
            "textComponent": "span",
          }
        }
        keyboardFocusedClassName="MuiButton-keyboardFocused-9"
        type="button"
      >
        <ButtonBase
          centerRipple={false}
          className="MuiButton-root-1"
          classes={
            Object {
              ".MuiButtonBase-root-15::-moz-focus-inner": "MuiButtonBase-root-15::-moz-focus-inner",
              "disabled": "MuiButtonBase-disabled-16",
              "root": "MuiButtonBase-root-15",
            }
          }
          disableRipple={false}
          disabled={false}
          focusRipple={true}
          intl={
            Object {
              "defaultFormats": Object {},
              "defaultLocale": "en",
              "formatDate": [Function],
              "formatHTMLMessage": [Function],
              "formatMessage": [Function],
              "formatNumber": [Function],
              "formatPlural": [Function],
              "formatRelative": [Function],
              "formatTime": [Function],
              "formats": Object {},
              "formatters": Object {
                "getDateTimeFormat": [Function],
                "getMessageFormat": [Function],
                "getNumberFormat": [Function],
                "getPluralFormat": [Function],
                "getRelativeFormat": [Function],
              },
              "locale": "en",
              "messages": Object {},
              "now": [Function],
              "textComponent": "span",
            }
          }
          keyboardFocusedClassName="MuiButton-keyboardFocused-9"
          tabIndex={0}
          type="button"
        >
          <button
            className="MuiButtonBase-root-15 MuiButton-root-1"
            disabled={false}
            intl={
              Object {
                "defaultFormats": Object {},
                "defaultLocale": "en",
                "formatDate": [Function],
                "formatHTMLMessage": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatPlural": [Function],
                "formatRelative": [Function],
                "formatTime": [Function],
                "formats": Object {},
                "formatters": Object {
                  "getDateTimeFormat": [Function],
                  "getMessageFormat": [Function],
                  "getNumberFormat": [Function],
                  "getPluralFormat": [Function],
                  "getRelativeFormat": [Function],
                },
                "locale": "en",
                "messages": Object {},
                "now": [Function],
                "textComponent": "span",
              }
            }
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label-3"
            >
              <FormattedMessage
                id="save"
                values={Object {}}
              >
                <span>
                  save
                </span>
              </FormattedMessage>
            </span>
            <withStyles(TouchRipple)
              center={false}
              innerRef={[Function]}
            >
              <TouchRipple
                center={false}
                classes={
                  Object {
                    "0%": "%",
                    "100%": "00%",
                    "50%": "0%",
                    "ripple": "MuiTouchRipple-ripple-21",
                    "rippleFast": "MuiTouchRipple-rippleFast-23",
                    "rippleVisible": "MuiTouchRipple-rippleVisible-22",
                    "root": "MuiTouchRipple-root-17",
                    "wrapper": "MuiTouchRipple-wrapper-18",
                    "wrapperLeaving": "MuiTouchRipple-wrapperLeaving-19",
                    "wrapperPulsating": "MuiTouchRipple-wrapperPulsating-20",
                  }
                }
              >
                <TransitionGroup
                  childFactory={[Function]}
                  className="MuiTouchRipple-root-17"
                  component="span"
                  enter={true}
                  exit={true}
                >
                  <span
                    className="MuiTouchRipple-root-17"
                  />
                </TransitionGroup>
              </TouchRipple>
            </withStyles(TouchRipple)>
          </button>
        </ButtonBase>
      </withStyles(ButtonBase)>
    </Button>
  </withStyles(Button)>
</ConditionalButton>
`;
